BEST_PRACTICES.md

```diff
@@ -0,0 +1,81 @@
+# Introduction
+This is a collection of tips, advice, gotchas and other best practices
+for using the [Jenkins Workflow
+plugin](https://github.com/jenkinsci/workflow-plugin/blob/master/README.md). Contributions
+and comments are happily accepted.
+
+# General tips and advice
+* Do everything that makes sense there within `stage`s. This makes your builds
```

Comment:
The close proximity of two "make" words causes this sentence to feel awkward to me. Suggested rewording: "This will make your builds easier to visualize, etc"